ray_on_golem/server/main.py

```diff
@@ -202,20 +207,58 @@ def start(port: int, registry_stats: bool, datadir: Optional[Path] = None):
     is_flag=True,
     help="Force the shutdown even if the webserver contains active nodes.",
 )
-def stop(port, force):
+@click.option(
+    "-k",
+    "--kill",
+    is_flag=True,
+    help="Kill the process if it fails to exit after shutdown.",
+)
+@click.option(
+    "--datadir",
+    type=Path,
+    help=f"Ray on Golem's data directory. By default, uses a system data directory: {DEFAULT_DATADIR}",
+)
+def stop(port, force, kill, datadir):
     from ray_on_golem.client import RayOnGolemClient
     from ray_on_golem.ctl import RayOnGolemCtl
-    from ray_on_golem.ctl.log import RayOnGolemCtlConsoleLogger
+    from ray_on_golem.ctl.log import RayOnGolemCtlConsoleLogger, RayOnGolemCtlError
     from ray_on_golem.server.models import ShutdownState
 
-    ctl = RayOnGolemCtl(RayOnGolemClient(port), RayOnGolemCtlConsoleLogger())
-    shutdown_state = ctl.stop_webserver(ignore_self_shutdown=True)
+    ctl = RayOnGolemCtl(
+        client=RayOnGolemClient(port), output_logger=RayOnGolemCtlConsoleLogger(), datadir=datadir
+    )
+
+    shutdown_state = ctl.stop_webserver(
+        ignore_self_shutdown=True, shutdown_delay=timedelta(seconds=0)
+    )
+
     if shutdown_state == ShutdownState.CLUSTER_NOT_EMPTY:
         if force or click.confirm("Force the shutdown?"):
-            shutdown_state = ctl.stop_webserver(
-                ignore_self_shutdown=True,
-                force_shutdown=True,
+            ctl.stop_webserver(
+                ignore_self_shutdown=True, force_shutdown=True, shutdown_delay=timedelta(seconds=0)
             )
+        else:
+            click.echo("Shutdown cancelled.")
+            return
+
+    try:
+        ctl.wait_for_stop(
+            starting_up=False,
+        )
+    except RayOnGolemCtlError as e:
+        click.echo(e)
+
+    proc = ctl.get_process_info()
+    if proc:
+        if kill or click.confirm("Terminate the webserver process?"):
+            try:
+                proc.kill()
+            except psutil.NoSuchProcess:
+                pass
+            ctl.clear_pid()
+            click.echo("Process terminated.")
+        else:
+            click.echo("The webserver is still running...")
```

Comment:
We can fold few indents here to improve readability

e.g.
```suggestion
    proc = ctl.get_process_info()
    if not proc:
        return
        
    if kill or click.confirm("Terminate the webserver process?"):
        ctl.kill_and_clear_pid() # new method
        click.echo("Process terminated.")
    else:
        click.echo("The webserver is still running...")
```

Reply:
good ideas, though I left the kill in place here...